src/eom-properties-dialog.c

```diff
@@ -165,15 +166,21 @@ pd_update_general_tab (EomPropertiesDialog *prop_dlg,
 
 	gtk_label_set_text (GTK_LABEL (prop_dlg->priv->bytes_label), bytes_str);
 
-	uri_str = eom_image_get_uri_for_display (image);
-	dir_str = g_path_get_dirname (uri_str);
-	gtk_label_set_text (GTK_LABEL (prop_dlg->priv->location_label),
-			    dir_str);
+	parent_file = g_file_get_parent (file);
+	if (parent_file == NULL) {
+		/* file is root directory itself */
+		parent_file = g_object_ref (file);
+	}
+	dir_str = g_file_get_basename (parent_file);
```

Comment:
Could we use `g_file_get_path` here? If I have an image in `/home/vkareh/Pictures/2018/07/16/foo.jpg`, the button displays just "16", which is not very useful. Using `g_file_get_path` gives you the full path as the label.

Reply:
So , `dir_str = g_file_get_path (parent_file);` ?